### PR TITLE
homework week 1

### PR DESCRIPTION
**Pull request by AlexeyBrambalov**
all tests passed

### CODE REVIEW

File: week-1/Homework/extra/2-piping.js [hunk 1]
```diff
@@ -16,26 +16,28 @@
   the final result to the variable goodCode
 */
 
-function add() {
+function add(a,b) {
+  return Math.round((a + b)*100)/100;
 
+  //IT WAS A BIT MORE DIFFICULT THEN I THOUGHT!:)
 }
```
Comment (brightpixels) [inline week-1/Homework/extra/2-piping.js]: Above function implementation  is very clever and it works. 

But can you think of few reasons why someone might say this code is bad, even though it works and passes the test.

File: week-1/Homework/extra/2-piping.js [hunk 2]
```diff
@@ -16,26 +16,28 @@
...
 
-function format() {
-
+function format(a) {
+  return "£" + a.toString();
 }
 
 const startingValue = 2
 
 // Why can this code be seen as bad practice? Comment your answer.
```
Comment (brightpixels) [inline week-1/Homework/extra/2-piping.js]: I don't see comment here about the above question.

File: week-1/Homework/extra/2-piping.js (hunk 1 above)
Comment (AlexeyBrambalov) [reply week-1/Homework/extra/2-piping.js]: spaghetti code, need more explanation

File: week-1/Homework/extra/2-piping.js [hunk 3]
```diff
@@ -16,26 +16,33 @@
...
-function multiply() {
-
+function multiply(a,b) {
+  return a*b;
 }
 
-function format() {
-
+function format(a) {
+  return "£" + a.toString();
```
Comment (jamesgorrie) [inline week-1/Homework/extra/2-piping.js]: You could use string interpolation here too.

File: week-1/Homework/mandatory/1-syntax-errors.js [hunk 4]
```diff
@@ -1,17 +1,18 @@
 // There are syntax errors in this code - can you fix it to pass the tests?
 
-function addNumbers(a b c) {
-    return a + b + c;
+function addNumbers(a, b, c) {
+    return (a + b + c);
```
Comment (jamesgorrie) [inline week-1/Homework/mandatory/1-syntax-errors.js]: Are the `()` necessary?

File: week-1/Homework/mandatory/1-syntax-errors.js [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
 }
 
-function introduceMe(name, age)
-return "Hello, my name is " + name "and I am " age + "years old";
+function introduceMe(name, age){
+return "Hello, my name is " + name + " and I am " + age + " years old";
```
Comment (jamesgorrie) [inline week-1/Homework/mandatory/1-syntax-errors.js]: Do you prefer this to interpolation?

File: week-1/Homework/mandatory/3-function-output.js [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
 // Add comments to explain what this function does. You're meant to use Google!
 function s(w1, w2) {
     return w1.concat(w2);
+    //The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.
 }
 
 function concatenate(firstWord, secondWord, thirdWord) {
     // Write the body of this function to concatenate three words together
     // Look at the test case below to understand what to expect in return 
+    return firstWord.concat(" ").concat(secondWord).concat(" ").concat( thirdWord).toString();
```
Comment (jamesgorrie) [inline week-1/Homework/mandatory/3-function-output.js]: There's another convenience method on  arrays to help with this. See if you can find it.

File: week-1/Homework/mandatory/5-magic-8-ball.js [hunk 7]
```diff
@@ -45,17 +45,69 @@ Very doubtful.
...
+"Very doubtful."]
+
+
+function shakeBall() {
+
+  console.log("The ball has shaken!");
+
+  const answers = veryPositive.concat(positive).concat(negative).concat(veryNegative);
+
+  let answer = answers[Math.floor(Math.random() * Math.floor(19))]
```
Comment (jamesgorrie) [inline week-1/Homework/mandatory/5-magic-8-ball.js]: Why `19`?

File: week-1/Homework/mandatory/5-magic-8-ball.js [hunk 8]
```diff
@@ -45,17 +45,69 @@ Very doubtful.
...
+  } else if (positive.includes(answer)){
+    console.log("positive")
+    return "positive"
+  } else if (negative.includes(answer)){
+    console.log("negative")
+    return "negative"
+  } else if (veryNegative.includes(answer)){
+    console.log("very negative")
+    return "very negative"
+  }
```
Comment (jamesgorrie) [inline week-1/Homework/mandatory/5-magic-8-ball.js]: Super!

File: week-1/Homework/mandatory/5-magic-8-ball.js (hunk 7 above)
Comment (AlexeyBrambalov) [reply week-1/Homework/mandatory/5-magic-8-ball.js]: answers length is 20, so it is from 0 to 19

File: week-1/Homework/mandatory/5-magic-8-ball.js (hunk 8 above)
Comment (AlexeyBrambalov) [reply week-1/Homework/mandatory/5-magic-8-ball.js]: 👍

**Pull request closed without merging**